### PR TITLE
Update links to GitHub repos moved to IntersectMBO

### PR DESCRIPTION
**Pull request by rphair**
Fixes https://github.com/cardano-foundation/developer-portal/issues/1182.

In general, redirections can't be relied upon forever and tend to fail _after_ the freshest knowledge behind the changes has moved elsewhere & I don't want to see this happen to this part of the Cardano documentation base.  Also there are some CIs that are confounded by invalid links, as was recently pointed out by @klntsky incidentally to https://github.com/cardano-foundation/developer-portal/issues/1258.

There were also some outright broken links, especially to asset files, that are cleaned up here: especially in relation to the Ledger spec (cc @lehins @WhatIsRT).

I've been meaning to get to this for the few months since https://github.com/cardano-foundation/developer-portal/issues/1182#issuecomment-1872300194 but the mapping I thought was possible in https://github.com/cardano-foundation/developer-portal/pull/1213#pullrequestreview-1887194099 wasn't possible to straightforwardly automate.

Without a list of `input-output-hk` repositories that were _not_ moved to `IntersectMBO` it was a matter of exhaustive search for the URLs mentioned here on the Dev Portal & seeing which ones were _not_ redirected.  After skipping the non-documentation parts of the Dev Portal in the first pathname list, I gradually added all these I could find in the second repository list (running this at the `developer-portal` repository root):

```
find . \( \
  -path ./node_modules -prune -o \
  -path ./.yarn -prune -o \
  -path ./.docusaurus -prune -o \
  -path ./.git -prune -o \
  -path ./docs/governance/cardano-improvement-proposals -prune -o \
  -path ./docs/governance/cardano-problem-statements -prune -o \
  -path ./build -prune \
\) -o -print0 | xargs -0 grep -n github.com.input-output-hk | \
sed -e s,input-output-hk/marlowe,, | \
sed -e s,input-output-hk/marlowe-cardano,,g | \
sed -e s,input-output-hk/marlowe-lambda,, | \
sed -e s,input-output-hk/real-world-marlowe,, | \
sed -e s,input-output-hk/MIPs,, | \
sed -e s,input-output-hk/purescript-marlowe,, | \
sed -e s,input-output-hk/daedalus,, | \
sed -e s,input-output-hk/lace,, | \
sed -e s,input-output-hk/metadata-registry-testnet,, | \
sed -e s,input-output-hk/offchain-metadata-tools,, | \
sed -e s,input-output-hk/cardano-js-sdk,, | \
sed -e s,input-output-hk/technical-docs,, | \
sed -e s,input-output-hk/implementation-decisions,, | \
sed -e s,input-output-hk/js-cardano-wasm,, | \
sed -e s,input-output-hk/cardano-node-wiki,, | \
sed -e s,input-output-hk/cardano-rest,, | \
sed -e s,input-output-hk/Developer-Experience-working-group,, | \
sed -e s,input-output-hk/haskell-course,, | \
sed -e s,input-output-hk/plutus-pioneer-program,, | \
cat > /tmp/io
```
... and then checking for all highlighted repos not nulled out by the series of `sed` commands...
```
grep --color=auto input-output-hk /tmp/io
```
... until making all the changes in this PR resulted in the output file being empty except for a code artefact.  The CIPs and CPSs are skipped because 1) I think we already have these weeded out & 2) I'm just submitting this PR for material sourced from the Dev Portal.  cc @Ryun1 @Crypto2099

Incidentally I think it might make sense with all these repo changes to retroactively add Intersect MBO to the [Contributors page](https://developers.cardano.org/docs/portal-contributors) (@katomm @Ryun1 ?).

### CODE REVIEW

**Review by os11k — APPROVED** [review]

File: docs/get-started/cardano-serialization-lib/overview.md [hunk 1]
```diff
@@ -64,9 +64,9 @@ following:
 
 Here are the location of the original [CDDL](http://cbor.io/tools.html) specifications:
 
--  Byron: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/byron/cddl-spec)
--  Shelley: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/shelley/test-suite/cddl-files)
--  Mary: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/shelley-ma/test-suite/cddl-files)
+-  Byron: [link](https://github.com/IntersectMBO/cardano-ledger/tree/master/eras/byron)
+-  Shelley: [link](https://github.com/IntersectMBO/cardano-ledger/tree/master/eras/shelley)
```
Comment (Ryun1) [inline docs/get-started/cardano-serialization-lib/overview.md:68]:
```suggestion
-  Shelley: [link](https://github.com/IntersectMBO/cardano-ledger/tree/master/eras/shelley/impl)
```

File: docs/get-started/cardano-serialization-lib/overview.md [hunk 2]
```diff
@@ -64,9 +64,9 @@ following:
 
 Here are the location of the original [CDDL](http://cbor.io/tools.html) specifications:
 
--  Byron: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/byron/cddl-spec)
--  Shelley: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/shelley/test-suite/cddl-files)
--  Mary: [link](https://github.com/input-output-hk/cardano-ledger/tree/master/eras/shelley-ma/test-suite/cddl-files)
+-  Byron: [link](https://github.com/IntersectMBO/cardano-ledger/tree/master/eras/byron)
```
Comment (Ryun1) [inline docs/get-started/cardano-serialization-lib/overview.md:67]:
```suggestion
-  Byron: [link](https://github.com/IntersectMBO/cardano-ledger/tree/master/eras/byron/impl)
```

File: docs/get-started/cardano-serialization-lib/overview.md (hunk 1 above)
Comment (Ryun1) [reply docs/get-started/cardano-serialization-lib/overview.md:68]: align with Mary link, and also is probably more useful to go into the `impl` director

**Review by rdlrt — APPROVED** [review]

**Review by katomm — APPROVED** [review]
This is a great contribution @rphair.  Where I have to disagree strongly is "to retroactively add Intersect MBO" to the contributors list. But I think that's a problem of interpretation here: the contributors list is to honour **substantial contributions from either people or organisations to the developer portal itself.** This does not change by moving some repos :))

**Review by katomm — APPROVED** [review]

**Pull request merged**